scripts/firedrake-status

```diff
@@ -0,0 +1,134 @@
+#! /usr/bin/env python
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+from pprint import pformat
+import logging
+import sys
+import os
+import subprocess
+from collections import OrderedDict, defaultdict
+
+parser = ArgumentParser(description="""Provide information on the currently downloaded version of Firedrake and its configuration.
+This is particularly useful information to include when reporting bugs.""",
+                        formatter_class=RawDescriptionHelpFormatter)
+parser.add_argument("--log", action='store_true',
+                    help="Log the output of the script to firedrake-status.log as well as to the console.")
+
+
+args = parser.parse_args()
+
+# Set up logging
+if args.log:
+    logging.basicConfig(level=logging.DEBUG,
+                        format='%(asctime)s %(levelname)-6s %(message)s',
+                        filename='firedrake-status.log',
+                        filemode='w')
+    console = logging.StreamHandler()
+    console.setLevel(logging.WARNING)
+    formatter = logging.Formatter('%(message)s')
+    console.setFormatter(formatter)
+    logging.getLogger().addHandler(console)
+else:
+    logging.basicConfig(level=logging.WARNING,
+                        format='%(message)s')
+log = logging.getLogger()
+
+
+def check_output(args, env=None):
+    return subprocess.check_output(args, stderr=subprocess.STDOUT, env=env)
+
+
+def quit(message):
+    log.error(message)
+    sys.exit(1)
+
+
+try:
+    firedrake_env = os.environ["VIRTUAL_ENV"]
+except KeyError:
+    quit("Unable to retrieve virtualenv name from the environment.\n Please ensure the virtualenv is active before running firedrake-update.")
+
+
+try:
+    import firedrake_configuration
+except ImportError:
+    log.info("Failed import firedrake_configuration module")
+    config = defaultdict(dict)
+else:
+    config = firedrake_configuration.get_config()
+
+try:
+    config["system"] = check_output(["uname", "-a"])
+except subprocess.CalledProcessError:
+    log.error("Failed to retrieve system information.")
+
+
+print "Firedrake Configuration:"
+for key, val in config["options"].iteritems():
+    print "    {}: {}".format(key, val)
+
+print "Additions:"
+if config["additions"]:
+    for a in config["additions"]:
+        print "   " + a
+else:
+    print "    None"
```

Comment:
When we couldn't import `firedrake_configuration` this prints as:

```
Firedrake Configuration:
Additions:
    None
```

Should we print something like:

```
Firedrake Configuration:
    No install configuration found
...
```

Or is that too confusing?